src/components/jupe-popover-bug/jupe-popover-bug.tsx

```diff
@@ -0,0 +1,30 @@
+import { Component, h } from "@stencil/core";
+
+@Component({
+  tag: "jupe-popover-bug",
+  styleUrl: "jupe-popover-bug.css",
+  shadow: true
+})
+export class JupePopoverBug {
+  render() {
+    return (
+      <calcite-popover-manager auto-close>
+        <calcite-input value="#00ff00">
+          <calcite-button appearance="outline" id="color-picker-toggle-button" slot="action">
+            <div class="swatch" style={{ "background-color": "#00ff00" }}>
+              &nbsp;
+            </div>
+          </calcite-button>
+        </calcite-input>
+        <calcite-popover
+          disablePointer={true}
+          label="color picker"
+          placement="auto"
+          referenceElement="color-picker-toggle-button"
```

Comment:
@mjuniper can you try setting the referenceElement as an element instead of an id string and see if that works?

```
toggleButtonEl: HTMLCalciteButtonElement;
```

```
<calcite-button appearance="outline" ref={(el) => this.toggleButtonEl = el)} slot="action">
```

```
referenceElement={this.toggleButtonEl}
```

Reply:
> can you try setting the referenceElement as an element instead of an id string and see if that works?

@driskull I tried that and it does not work. _But_ that approach does not work even with `shadow: false` so maybe I did it wrong (I tried a bunch of variations)?

Reply:
You shouldn't need to set this part on the popover...

```
ref={this.setPopoverElement.bind(this)}
```

You can change the button to this...

```
<calcite-button
  appearance="outline"
  ref={(el) => (this. toggleButtonEl = el)}
  slot="action"
>
```

Also, I think that the `toggleButtonEl` property needs to have the `@State()` modifier so that it knows to rerender when it is set.

Let me know if that works. setting shadow to false should't be necessary.

Reply:
I know I should not need to do that part on the popover, that was just me trying to get it to work. But your suggestion of  decorating `toggleButtonEl` with `@State()` is better.

That works, but only with `shadow: false`.

The entirety of the component is now:

```
import { Component, h, State } from "@stencil/core";

@Component({
  tag: "jupe-popover-bug",
  styleUrl: "jupe-popover-bug.css",
  shadow: true
})
export class JupePopoverBug {
  @State()
  toggleButtonEl: HTMLCalciteButtonElement;

  render() {
    return (
      <calcite-popover-manager auto-close>
        <calcite-input value="#00ff00">
          <calcite-button
            appearance="outline"
            ref={el => (this.toggleButtonEl = el)}
            slot="action"
          >
            <div class="swatch" style={{ "background-color": "#00ff00" }}>
              &nbsp;
            </div>
          </calcite-button>
        </calcite-input>
        <calcite-popover
          disablePointer={true}
          label="color picker"
          placement="auto"
          referenceElement={this.toggleButtonEl}
        >
          <calcite-color-picker allow-empty="true" format="hex" scale="m" value="#00ff00" />
        </calcite-popover>
      </calcite-popover-manager>
    );
  }
}
```

And that will not work unless you change it to `shadow: false` (or comment that line).